core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -82,6 +82,33 @@ default ConnectorTableHandle getTableHandle(ConnectorSession session, SchemaTabl
         return null;
     }
 
+    /**
+     * Returns a table handle for the specified table name and version, or {@code null} if {@code tableName} relation does not exist
+     * or is not a table (e.g. is a view, or a materialized view).
+     *
+     * @throws TrinoException implementation can throw this exception when {@code tableName} refers to a table that
+     * cannot be queried.
+     * @see #getView(ConnectorSession, SchemaTableName)
+     * @see #getMaterializedView(ConnectorSession, SchemaTableName)
+     */
+    @Nullable
+    default ConnectorTableHandle getTableHandle(
+            ConnectorSession session,
+            SchemaTableName tableName,
+            Optional<ConnectorTableVersion> startVersion,
+            Optional<ConnectorTableVersion> endVersion)
+    {
+        ConnectorTableHandle tableHandle = getTableHandle(session, tableName);
+        if (tableHandle == null) {
```

Comment:
So this is a change in default behavior, is it intentional? Previously, an error would always be thrown for connectors that did not implement the time travel SPIs, now that error can be skipped if the table handle does not exist.

Reply:
> So this is a change in default behavior, is it intentional?

Yes, it's intentional.

For a query `... FROM TABLE non_existent_table AS OF ...`
returning "table not found" seems more reasonable to me.
If a table doesn't exist, there is no point in determining which connector it would come from and whether it would support versioning.